x-pack/plugins/endpoint/public/embeddables/resolver/view/index.tsx

```diff
@@ -31,35 +34,57 @@ const StyledGraphControls = styled(GraphControls)`
 `;
 
 export const Resolver = styled(
-  React.memo(function Resolver({ className }: { className?: string }) {
+  React.memo(function Resolver({
+    className,
+    selectedEvent,
+  }: {
+    className?: string;
+    selectedEvent?: LegacyEndpointEvent;
+  }) {
     const { processNodePositions, edgeLineSegments } = useSelector(
       selectors.processNodePositionsAndEdgeLineSegments
     );
 
+    const dispatch: (action: ResolverAction) => unknown = useDispatch();
     const { projectionMatrix, ref, onMouseDown } = useCamera();
+    const isLoading = useSelector(selectors.isLoading);
 
+    useEffect(() => {
```

Comment:
try `useLayoutEffect`. This way, if the resolver is rendered with a selected event initially, it'll dispatch the event and rerender before being flushed to the DOM. Otherwise you'll have a flash of uninitialized resolver DOM.